audformat/core/database.py

```diff
@@ -433,6 +441,398 @@ def duration(file: str) -> pd.Timedelta:
+            elif isinstance(scheme.labels, dict) and not strict:
+                labels = pd.DataFrame.from_dict(
+                    scheme.labels,
+                    orient='index',
+                )
+                if requested_scheme in labels:
+                    scheme_mappings.append((scheme_id, requested_scheme))
+
+        # --- Get data for requested schemes
+        ys = []
+        for table_id in tables:
+
+            # Handle non-existing tables
+            if table_id not in self.tables:
+                continue
+            else:
+                table = self[table_id]
+
+            # Limit search by split
+            if splits is not None and table.split_id not in splits:
+                continue
+
+            for column_id, column in table.columns.items():
+                y = None
+                # Scheme directly stored in column
+                if scheme_in_column(requested_scheme, column, column_id):
+                    y = self[table_id][column_id].get()
+                    y.name = requested_scheme
+                    append_series(ys, y, column_id)
+                # Get series based on label of scheme
+                else:
+                    for (scheme_id, mapping) in scheme_mappings:
+                        if scheme_in_column(scheme_id, column, column_id):
+                            if column.scheme_id is None:
+                                y = empty_series(requested_scheme)
+                            else:
+                                y = self[table_id][column_id].get(map=mapping)
+                                y.name = requested_scheme
+                            append_series(ys, y, column_id)
+
+        # --- Ensure we have a common dtype for requested scheme
+        categorical_dtypes = [
+            y.dtype for y in ys
+            if isinstance(y.dtype, pd.CategoricalDtype)
+        ]
+        dtypes_of_categories = [
+            dtype.categories.dtype
+            for dtype in categorical_dtypes
+        ]
+        if len(categorical_dtypes) > 0:
+            if len(set(dtypes_of_categories)) > 1:
+                # Don't know if this can ever be triggered,
+                # but make sure we raise the same error as
+                # pd.api.types.union_categoricals
+                raise TypeError(  # pragma: nocover
+                    'dtype of categories must be the same'
+                )
+            dtype = dtypes_of_categories[0]
+            # Convert everything to categorical data
+            for n, y in enumerate(ys):
+                if not isinstance(y.dtype, pd.CategoricalDtype):
+                    ys[n] = y.astype(
+                        pd.CategoricalDtype(
+                            y.array.dropna().unique().astype(dtype)
+                        )
+                    )
+            # Find union of categorical data
+            data = [y.array for y in ys]
+            data = pd.api.types.union_categoricals(data)
```

Comment:
I am encountering the data type error already discussed when the same schemes returns mixed types. 

The comment was this:

>> To me it sounds fine if we raise an error if we get different data types for the same scheme. Maybe we can explain in the error message to the user how she should fix the database. 


The steps for me to reproduce were this:

```python 

    db = audb.load(
        'mozillacommonvoice',
        version='10.3.0',
        only_metadata=True,
        verbose=False,
        full_path=False,
    )

data = db.get('age') 

```

Schemes conflicting  were 

```
$ys[2].array                                                                                                                                                                                                                                                    
[55, 55, 25, 25, 45, ..., 25, 25, 25, 25, 25]                                                                                                                                                                                                                        
 Length: 85986                                                                                                                                                                                                                                                        
 Categories (8, int64): [55, 25, 45, 35, 15, 65, 75, 85]                                                                                                                                                                                                              
$ ys[3].array                                                                                                                                                                                                                                                    
['fourties', 'fourties', 'fourties', 'fourties', 'fourties', ..., 'thirties', 'thirties', 'thirties', 'thirties', 'thirties']                                                                                                                                        
Length: 8375                                                                                                                                                                                                                                                         
Categories (9, object): ['teens', 'twenties', 'thirties', 'fourties', ..., 'sixties', 'seventies',                                                                                                                                                                   
                         'eighties', 'nineties']                                                                                                                                                                                                                     

```

My question is: How could the suggestion for the user look like? 

The categories in the database are not easily reconciled (the int typed one does not really allow mapping to the object typed one).

Reply:
I have updated the error message and your example now results in:

```
...
TypeError: Cannot join labels for scheme 'age' with different data types: int64, object
```